internal/statemachine/state_machine.go

```diff
@@ -301,7 +306,7 @@ func (stateMachine *StateMachine) postProcessGadgetYaml() error {
 		}
 	}
 
-	if !rootfsSeen && len(stateMachine.GadgetInfo.Volumes) == 1 {
+	if !farthestOffsetUnknown && !rootfsSeen && len(stateMachine.GadgetInfo.Volumes) == 1 {
```

Comment:
This is only for very old stuff only. The partition is normally defined. Instead of this condition, if we detected we need to define that partition and `farthestOffset` was not not calculated, then maybe we should give an error instead of silently ignore it.